src/Analyzers/Core/Analyzers/RemoveRedundantEqualityWithTrue/RemoveRedundantEqualityWithTrueDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.RemoveRedundantEqualityWithTrue
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    internal sealed class RemoveRedundantEqualityWithTrueDiagnosticAnalyzer
+        : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        private static readonly LocalizableResourceString s_title =
+            new(nameof(AnalyzersResources.Remove_redundant_equality_with_true), AnalyzersResources.ResourceManager, typeof(AnalyzersResources));
+
+        public RemoveRedundantEqualityWithTrueDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.RemoveRedundantEqualityWithTrueDiagnosticId,
+                  option: null,
+                  s_title)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
```

Comment:
I'm not sure of the category.

Reply:
this is correct :)

Reply:
it basically means: if a method body is edited, do i only have to analyze just that method body.